packages/vue-grid/src/Grid.ts

```diff
@@ -0,0 +1,43 @@
+function hForVue3(instance: any) {
+  return function h(type: string, props: any, children: any[]) {
+    let ref = null;
+
+    if (props.ref) {
+      ref = {
+        i: instance.$,
+        r: props.ref,
+      };
+    }
+    return {
+      shapeFlag: 17,
```

Comment:
How about using [ShapeFlags](https://github.com/vuejs/vue-next/blob/master/packages/shared/src/shapeFlags.ts) from `@vue/shared` instead?

```ts
import { ShapeFlags } from "@vue/shared";

const shapeFlag = ShapeFlags.ARRAY_CHILDREN | ShapeFlags.ELEMENT;
```

Reply:
@WoodNeck 

shapeflag exists only as a type in shared.

Reply:
Did you tested that?

https://github.com/vuejs/vue-next/blob/b46d8578fd7e0cb7f82620d21f2a0ad1094c34af/packages/runtime-core/src/componentRenderUtils.ts#L62

They're using it as a value in their code.

Reply:
I also checked the code. There are no ShapeFlags in the transformed code.

Reply:
That's weird, but okay then.

As it's a magic number, please change that whenever possible.
Or, at least leave a comment with a link to https://github.com/vuejs/vue-next/blob/master/packages/shared/src/shapeFlags.ts there

Reply:
@WoodNeck 

Okay!